<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>detector_panel_buffer</class>
 <widget class="QDialog" name="detector_panel_buffer">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>478</width>
    <height>157</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configure panel buffer</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="4" column="0">
    <widget class="QTabWidget" name="tab_widget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="border_tab">
      <attribute name="title">
       <string>Border</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="3" column="1">
        <widget class="QSpinBox" name="border_y_spinbox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100000</number>
         </property>
         <property name="value">
          <number>1</number>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QSpinBox" name="border_x_spinbox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100000</number>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="border_x_label">
         <property name="text">
          <string>X Border (mm):</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="border_y_label">
         <property name="text">
          <string>Y Border (mm):</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="numpy_tab">
      <attribute name="title">
       <string>Numpy</string>
      </attribute>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLineEdit" name="file_name"/>
       </item>
       <item>
        <widget class="QPushButton" name="select_file_button">
         <property name="text">
          <string>Select NumPy Array File</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QLabel" name="config_mode_label">
       <property name="text">
        <string>Configuration Mode:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="config_mode">
       <property name="currentText">
        <string>Border</string>
       </property>
       <item>
        <property name="text">
         <string>Border</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>NumPy</string>
        </property>
       </item>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="5" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QDialogButtonBox" name="button_box">
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
       <property name="centerButtons">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="modesLabel">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>The panel buffer can be set in two ways. Either by provided the border in mm or through a NumPy array, interpretation is boolean with true marking valid pixels.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tab_widget</tabstop>
  <tabstop>file_name</tabstop>
  <tabstop>select_file_button</tabstop>
  <tabstop>border_x_spinbox</tabstop>
  <tabstop>border_y_spinbox</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>button_box</sender>
   <signal>accepted()</signal>
   <receiver>detector_panel_buffer</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>button_box</sender>
   <signal>rejected()</signal>
   <receiver>detector_panel_buffer</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>217</x>
     <y>205</y>
    </hint>
    <hint type="destinationlabel">
     <x>217</x>
     <y>113</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
